mitiq/zne/scaling/folding.py

```diff
@@ -289,6 +367,45 @@ def _get_num_to_fold(scale_factor: float, ngates: int) -> int:
 
 
 # Local folding functions
+@noise_scaling_converter
+def fold_all(
+    circuit: QPROGRAM,
+    num_folds: int = 1,
+    exclude: FrozenSet[Any] = frozenset(),
+) -> QPROGRAM:
+    """Returns a circuit with all gates folded locally.
+
+    Args:
+        circuit: Circuit to fold.
+        num_folds: Number of times to add G^dag G after each gate G.
+        exclude: Do not fold these gates. Supported gate keys are listed in
+            the following table.::
+
+                Gate key    | Gate
+                -------------------------
+                "H"         | Hadamard
+                "X"         | Pauli X
+                "Y"         | Pauli Y
+                "Z"         | Pauli Z
+                "I"         | Identity
+                "CNOT"      | CNOT
+                "CZ"        | CZ gate
+                "TOFFOLI"   | Toffoli gate
+                "single"    | All single qubit gates
+                "double"    | All two-qubit gates
+                "triple"    | All three-qubit gates
+    """
+    _check_foldable(circuit)
+
+    folded = deepcopy(circuit)
+    measurements = _pop_measurements(folded)
+
+    folded = _fold_all(folded, num_folds, exclude)
+
+    _append_measurements(folded, measurements)
+    return folded
```

Comment:
This function is not properly a noise scaling function since it does not take a `scale_factor` as an input. So it should not be directly used as an argument of `execute_with_zne` .  Maybe we could:

- Add a note to warn the user about this problem.
or
- Change the function to 
```
@noise_scaling_converter
def uniform_scaling(
    circuit: QPROGRAM,
    scale_factor: int,
    exclude: FrozenSet[Any] = frozenset(),
) -> QPROGRAM:
    """Returns a circuit with all gates folded locally. This uniform
    folding method can effectively scale the noise by odd integers.
    Args:
        circuit: Circuit to fold.
        scale_factor: The noise scale factor. It must be an odd integer.
        exclude: Do not fold these gates. Supported gate keys are listed in
            the following table.::
                Gate key    | Gate
                -------------------------
                "H"         | Hadamard
                "X"         | Pauli X
                "Y"         | Pauli Y
                "Z"         | Pauli Z
                "I"         | Identity
                "CNOT"      | CNOT
                "CZ"        | CZ gate
                "TOFFOLI"   | Toffoli gate
                "single"    | All single qubit gates
                "double"    | All two-qubit gates
                "triple"    | All three-qubit gates
    """
    _check_foldable(circuit)

    folded = deepcopy(circuit)

    measurements = _pop_measurements(folded)
    
    if scale_factor is not np.close to an odd integer:
           raise error

    num_folds = int((scale_factor - 1) / 2)
    folded = _fold_all(folded, num_folds, exclude)

    _append_measurements(folded, measurements)
    return folded

Reply:
Ok I think you convinced me. It's a bit of a bummer we can't use `num_folds` here as it's so natural to think about and I've gotten this feedback multiple times. But as far as extrapolate to 0 is concerned we really need 0 to correspond to no noise, but `num_folds=0` is the base noise not zero noise.